src/sentry/monitors/endpoints/organization_monitor_details.py

```diff
@@ -111,6 +112,8 @@ def put(self, request: Request, organization, project, monitor) -> Response:
 
         if params:
             monitor.update(**params)
+            if params.get("status"):
+                MonitorEnvironment.objects.filter(monitor=monitor).update(status=params["status"])
```

Comment:
this will fix a sort bug by marking the `MonitorEnvironment` as `DISABLED`

Reply:
Are we sure we want this? I would think we would just want to keep the sorting logic so that monitors that are disabled get sorted to the bottom and just ignore the disabled in that case.

I would also probably not do this in this cleanup PR

Reply:
the issue is that we only read off the `MonitorEnvironment` status during sort right now, and that never gets marked as `DISABLED`. this is the only place we have that hook. some of this follow up work was going to simplify the sort logic but really it's more pending the "one row per monitor w/multiple environments" UI

Reply:
https://github.com/getsentry/sentry/pull/48039
fix is here. will remove this

Reply:
Good stuff 👍